BugOsPos.Api/Controllers/LocationsController.cs

```diff
@@ -0,0 +1,39 @@
+﻿using BugOsPos.Application.Locations;
+using BugOsPos.Contracts.Locations;
+using BugOsPos.Infrastructure.Authentication;
+using MapsterMapper;
+using BugOsPos.Domain.Common.ErrorsCollection;
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+
+namespace BugOsPos.Api.Controllers;
+
+[ApiController]
+public class LocationsController : ApiController
+{
+    private readonly IMapper _mapper;
+    private readonly ISender _mediator;
+
+    public LocationsController(IMapper mapper, ISender mediator)
+    {
+        _mapper = mapper;
+        _mediator = mediator;
+    }
+
+    [HttpGet("locations/{id}")]
+    public async Task<IActionResult> GetLocation(int id)
+    {
+        if (GetClaimValue(JwtSettings.FranchiseClaim) is not string franchiseIdString)
+            return Problem(new() { Errors.Authentication.FranchiseIdMissing });
+
+        if (!int.TryParse(franchiseIdString, out var franchiseId))
+            return Problem(new() { Errors.Authentication.InvalidFranchiseId });
+
+        var command = new GetLocationByIdQuery(id);
```

Comment:
This check's user's franchise.

But if you want to make sure that it is an employee making this command, you should add JwtSettings.EmployeeClaim check. Not sure if it's needed, since documentation is vague and contradicting.